pkg/i2gw/providers/openapi3/converter.go

```diff
@@ -0,0 +1,619 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package openapi3
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"slices"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/getkin/kin-openapi/openapi3"
+	"github.com/samber/lo"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	gatewayv1 "sigs.k8s.io/gateway-api/apis/v1"
+	gatewayv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+
+	"github.com/kubernetes-sigs/ingress2gateway/pkg/i2gw"
+	"github.com/kubernetes-sigs/ingress2gateway/pkg/i2gw/providers/common"
+)
+
+const (
+	HostWildcard   = "*"
+	HostSeparator  = ","
+	ParamSeparator = ","
+
+	HTTPRouteRulesMax      = 16
+	HTTPRouteMatchesMax    = 8
+	HTTPRouteMatchesMaxMax = HTTPRouteRulesMax * HTTPRouteMatchesMax
+)
+
+var uriRegexp = regexp.MustCompile(`^((https?)://([^/]+))?(/.*)?$`) // [_][1] = scheme, [_][2] = host, [_][3] = path
+
+type Converter interface {
+	Convert(Storage) (i2gw.GatewayResources, field.ErrorList)
+}
+
+// NewConverter returns a converter of OpenAPI Specifications 3.x from a storage into Gateway API resources.
+func NewConverter(conf *i2gw.ProviderConf) Converter {
+	converter := &converter{
+		namespace:    conf.Namespace,
+		tlsSecretRef: types.NamespacedName{},
+		backendRef:   toBackendRef(""),
+	}
+
+	if ps := conf.ProviderSpecific[ProviderName]; ps != nil {
```

Comment:
what ensures that all those flags are set if ps != nil?

Reply:
They don't have to. `toNamespacedName` and `toBackendRef` funcs will fallback to the defaults, which are analogous in effect to an empty `gatewayClassName`.

IOW, we need this nil check not to panic, but we don't need to check each individual flag.